frontend/src/components/Collections/components/Grid/components/CollectionsGrid/index.tsx

```diff
@@ -28,17 +33,29 @@ const CollectionsGrid: FC<Props> = ({ collections, accessType }) => {
         </tr>
       </thead>
       <tbody>
-        {collections?.map(({ id, visibility }) => (
-          <CollectionRow
-            id={id}
-            key={id + visibility}
-            visibility={visibility}
-            accessType={accessType}
-          />
-        ))}
+        {sortByUpdatedAtDescending(collections)?.reduce(
+          (acc, { id, visibility }) => {
+            if (!displayVisibility || visibility === displayVisibility) {
```

Comment:
Filter out any non-matching visibilities if applicable

Reply:
Actually we can use `sortByUpdatedAtDescending(collections)?.filter()` instead of `.reduce()` here for clarity, I believe!

E.g., 

```tsx
{ renderCollections(sortByUpdatedAtDescending(collections)) }

function renderCollections(collections = []) {
  if (!collections.length) return null;

  return collections.filter((collection) => isConditionMet(collections))
}
```